text/0000-project-descriptor-domains.md

```diff
@@ -0,0 +1,155 @@
+
+There will be a `api` key in this table to show which schema version to use for the overall file. It'll be bumped to `0.2` encapsulating the changes to this table as well as the domain namespace tables.
+
+## Reverse Domain Namespaced Tables
+All other tables will namespaced by reverse domains. With these new namespaces, a versioned schema CAN apply to each table.
+
+[build]` table will now be `[io.buildpacks.build]`. The keys under `[build]` will not change. The [schema validation RFC](https://github.com/buildpacks/rfcs/blob/main/text/0054-project-descriptor-schema.md) can now be applied just to our own `[io.buildpacks]` namespace. While not mandatory for other namespaced tables, the `[io.buildpacks]` will have belong in the spec.
+
+## Non Buildpacks Project Descriptor Example
+
+For a non-project buildpacks project adopting the [fly.io](https://fly.io) `fly.toml` taken from the [phoenix-liveview-clustoer example app](https://github.com/fly-apps/phoenix-liveview-cluster/blob/master/fly.toml):
+
+```TOML
+[_]
+api = "0.2"
+
+[io.fly]
+app = "liveview-counter"
+
+kill_signal = "SIGTERM"
+kill_timeout = 5
+
+[io.fly.experimental]
+private_network = true
+
+[[io.fly.services]]
+  internal_port = 4000
+  protocol = "tcp"
+
+  [io.fly.services.concurrency]
+    hard_limit = 25
+    soft_limit = 20
+
+  [[io.fly.services.ports]]
+    handlers = ["http"]
+    port = "80"
+
+  [[io.fly.services.ports]]
+    handlers = ["tls", "http"]
+    port = "443"
+
+  [[io.fly.services.tcp_checks]]
+    grace_period = "30s" # seems to take about 30s to boot
+    interval = "10s"
+    port = "8080"
+    restart_limit = 0
+    timeout = "2s"
+```
+
+## Backwards Compatibility
+
+The bundler project switched from YAML to a proprietary Ruby DSL for the `Gemfile` and was able to detect the differences to determine which version to use. Our parsers can detect for `[io.buildpacks]` key specifically to differentiate between the two. In the future, the API key under this table can be used.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+This is outlined in ["What it is"](#what-it-is).
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+There are two major drawbacks:
+
+- This is a backwards breaking change and the schemas aren't compatible at all.
+- TOML doesn't use whitespacing to determine nesting, so the full key is used which can be verbose.
+
+# Alternatives
+[alternatives]: #alternatives
+
+This suprsedes the [Project Descriptor Flexibility RFC](https://github.com/buildpacks/rfcs/pull/95).
+
+## Special Case Project Namespace
```

Comment:
Alternatively, we could define that the root level of this TOML does not use reverse domain keys and move them under a generic `config` (or similar) key. This would allow us to use descriptive keys for things that are generic (such as `[project]`) and still allow folks using the `.project` TLD to use this feature without workarounds.

This would also allow us to keep the `build` key if we'd like to keep it.

Example:
```
[project]
id = ""
name = ""
# ...

[config.com.example]
awesome = "sauce"
```

I'm not sure how much I personally like this, but it seems to be a viable alternative.